### ISSUE TITLE
`calcite-pagination` navigation buttons does not shift focus to the first focusable element

### ISSUE THREAD

**Issue by minhiu** [Esri/calcite-design-system#11920]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

**Issue**
There are interactive elements in page content that do not receive focus when the next/previous button in Pagination is activated.

**User Impact**
Screen reader users will be unable to determine that these page content has changed unless they review the entire page. Keyboard-only users will have difficulty navigating into the page content.

 
**Recommendation**
Ensure that when the "next page" button in a pagination component is activated, the focus moves to the first interactive element on the new page. Use the JavaScript focus() method to move keyboard focus to the first interactive element.

### Actual Behavior

- After hitting "Next" or "Previous" Pagination button, the focus stays the same.
- Keyboard users stay "stuck" there and might not realize the content has changed.
- Screen readers may not announce the new content unless focus moves or there's a screen reader cue.

### Expected Behavior

After loading new paginated content (say, a new list of results), programmatically move focus to the first interactive element (like a link or button) or to a heading that introduces the content.

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/pagination/

### Reproduction Steps

1. Locate the pagination component.
2. Navigate the page using tab key.
3. Activate the "next page" button to move to the next set of content.
4. Observe the behavior of the focus after the new page loads.

### Reproduction Version

3.0.0

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Business/Community Analyst
